spring-core/src/main/java/org/springframework/core/MethodParameter.java

```diff
@@ -67,6 +74,8 @@
 
 	private volatile Class<?> parameterType;
 
+	private volatile Boolean isKotlinNullable;
```

Comment:
I think I would try initially without caching the result, unless you have specific concern about the performances